query-connector/src/app/database-service.ts

```diff
@@ -48,45 +48,48 @@ export const getSavedQueryByName = async (name: string) => {
 };
 
 /**
- * Helper function to filter the valueset-mapped rows of results returned from
- * the DB for particular types of related clinical services.
- * @param vsItems A list of value sets mapped from DB rows.
- * @param type One of "labs", "medications", or "conditions".
- * @returns A list of rows containing only the predicate service type.
+ * Maps the results returned from the DIBBs value set and coding system database
+ * into a collection of value sets, each containing one or more Concepts build out
+ * of the coding information in the DB.
+ * @param rows The Rows returned from the DB Query.
+ * @returns A list of ValueSets, which hold the Concepts pulled from the DB.
  */
-export const filterValueSets = async (
-  vsItems: ValueSetItem[],
-  type: "labs" | "medications" | "conditions",
+export const mapQueryRowsToConceptValueSets = async (
+  rows: QueryResultRow[],
 ) => {
-  // Assign clinical code type based on desired filter
-  // Mapping is established in TCR, so follow that convention
-  let valuesetFilters = valueSetTypeToClincalServiceTypeMap[type];
-  const results = vsItems.filter((vs) =>
-    valuesetFilters.includes(vs.clinicalServiceType),
-  );
-  return results;
-};
+  // Create groupings of rows (each of which is a single Concept) by their ValueSet ID
+  const vsIdGroupedRows = rows.reduce((conceptsByVSId, r) => {
+    if (!conceptsByVSId[r["valueset_id"]]) {
```

Comment:
nit: could we do `if(!(r["valueset_id"] in conceptsByVSId))` here? Don't think this is very likely, but in the off chance that somehow the dictionary values get populated with something falsy, this might not behave like expected. The `in` keyword is made for cases like this specifically

Reply:
sure, seems fine to change